daemon/dao/etagdao.go

```diff
@@ -0,0 +1,54 @@
+package dao
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+)
+
+const (
+	Prefix     string        = "Etag"
+	timeToLive time.Duration = 3600000 * time.Millisecond
+)
+
+type EtagDao struct {
+	Ctx context.Context
+	Rdb *redis.Client
+}
+
+func (e *EtagDao) SingleCache(etag *Etag) {
+	key := Prefix + ":" + etag.Index
+	b, err := json.Marshal(etag)
+	if err != nil {
+		return
+	}
+
+	err = e.Rdb.Set(e.Ctx, key, string(b), timeToLive).Err()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (e *EtagDao) BatchCache(etags []*Etag) {
+	for _, etag := range etags {
```

Comment:
Need to use e.Rdb.MSet instead of using Set. Batch is one command with multiple values.